src/common/Preforker.h

```diff
@@ -108,7 +108,7 @@ class Preforker {
   int signal_exit(int r) {
     if (forked) {
       /* If we get an error here, it's too late to do anything reasonable about it. */
-      (void)safe_write(fd[1], &r, sizeof(r));
+      [[maybe_unused]] auto n = safe_write(fd[1], &r, sizeof(r));
```

Comment:
I'm not really a fan of what appears to be gcc/clang magic to do this but I leave it up to your judgement @tchaikov. Is there something wrong with what's in #21481?

Reply:
@batrick there is nothing wrong with #21481. and IMHO, this is not gcc/clang magic, it's an attribute specifier defined in C++17. i think, it's just more explicit and more readable than the good old `(void)` way.